### PR TITLE
Upgrade to rgbds 0.5.2, and use newer macro syntax

### PR DESCRIPTION
**Pull request by kemenaran**
Fixes future macro warnings on the (yet unreleased) rgbds 0.6, by using the newer macro syntax. Credits to @ShadowOne333

This PR doesn't include the "unmapped chars" warning appearing in RGBDS 0.6 for now. The warning cause should probably be fixed - but for now, it can be worked around by disabled the warning using the command line :

```shell
make ASFLAGS='--export-all -Wno-unmapped-char'
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -9,7 +9,7 @@
 2BPP    := rgbgfx
 
 ASM     := rgbasm
-ASFLAGS := --export-all
+ASFLAGS := --export-all -Wno-unmapped-char
```
Comment (daid) [inline Makefile:12]: It would be better to fix this instead of disabling the warning.

**Comment by kemenaran** [comment]
This PR doesn't include the "unmapped chars" warning appearing in RGBDS 0.6 for now. The warning cause should probably be fixed - but for now, it can be worked around by disabled the warning using the command line :

```shell
make ASFLAGS='--export-all -Wno-unmapped-char'
```

**Comment by ShadowOne333** [comment]
Thanks for the credit :D
I'll try to see if I can fix the unmapped-char warning fixed as well this week.
I'll post any changes here when I get to them.

**Comment by ShadowOne333** [comment]
Seems like the only way to fix the warning about the unmapped-char is to actually map or define each character from Link's Awakening manually.
A contributor to rgbds, Rangi, suggested the following:
```; map each character to its default ASCII value
DEF PRINTABLE_ASCII EQUS " !\"#$%&'()*+,-./0123456789:;<=>?@ABCDEFGHIJKLMNOPQRSTUVWXYZ[\\]^_`abcdefghijklmnopqrstuvwxyz\{|}~"
for i, STRLEN("{PRINTABLE_ASCII}")
    charmap STRSUB("{PRINTABLE_ASCII}", i + 1, 1), i + $20
endr```

That would solve the warning, all that'd be needed is to properly put the corresponding characters in order.
Here's another suggested way of doing it (with \t \n \r):
```; map each character to its default ASCII value
DEF PRINTABLE_ASCII EQUS "\t\n\r !\"#$%&'()*+,-./0123456789:;<=>?@ABCDEFGHIJKLMNOPQRSTUVWXYZ[\\]^_`abcdefghijklmnopqrstuvwxyz\{|}~"
for i, 1, STRLEN("{PRINTABLE_ASCII}") + 1
    charmap STRSUB("{PRINTABLE_ASCII}", i, 1), (STRSUB("{PRINTABLE_ASCII}", i, 1))
endr```

**Comment by kemenaran** [comment]
I guess that piece of code could go into src/constants/charmaps/main.asm 

I'm open to both ways: either defining all characters by hand, or using a macro with a comment like "; Explicitely define ASCII characters".

**Comment by ShadowOne333** [comment]
Mmmm I tried adding all of the ASCII table to the charmaps `main.asm` for each language, but I am still getting the warnings when compiling with the dialogue and credits files.
You can check the main.asm from src/constants/main.asm in the zip attached.

[main.zip](https://github.com/zladx/LADX-Disassembly/files/9567806/main.zip)

**Comment by ShadowOne333** [comment]
I found a hacky way to remove the warnings, I ended up including the charmaps inside each of the credits files for each language, and then for the French and German languages, I added 4 specific charmaps so they match properly.

Now when compiling, it shouldn't throw any errors by using the files in this zip:
[charmaps.zip](https://github.com/zladx/LADX-Disassembly/files/9579044/charmaps.zip)

I have included both the main.asm files for each language that were modified, as well as all the credits.asm files modified too.
Compiling shouldn't throw any kind of `unmapped-char` errors now.
If everything goes well, I think this alongside the macro changes should be good for committing.

**Comment by kemenaran** [comment]
Alright, I'll first merge this PR, and then open another one for the charmaps.

**Pull request merged**